src/nvim/fileio.c

```diff
@@ -4333,108 +4333,95 @@ void shorten_fnames(int force)
   redraw_tabline = TRUE;
 }
 
-/*
- * add extension to file name - change path/fo.o.h to path/fo.o.h.ext
- *
- * Assumed that fname is a valid name found in the filesystem we assure that
- * the return value is a different name and ends in 'ext'.
- * "ext" MUST be at most 4 characters long if it starts with a dot, 3
- * characters otherwise.
- * Space for the returned name is allocated, must be freed later.
- * Returns NULL when out of memory.
- */
-char_u *
-modname (
-    char_u *fname,
-    char_u *ext,
-    int prepend_dot                /* may prepend a '.' to file name */
-)
+/// add extension to file name - change path/fo.o.h to path/fo.o.h.ext
+///
+/// Assumed that fname is a valid name found in the filesystem we assure that
+/// the return value is a different name and ends in 'ext'.
+/// "ext" MUST be at most 4 characters long if it starts with a dot, 3
+/// characters otherwise.
+/// Space for the returned name is allocated, must be freed later.
+/// Returns NULL when out of memory.
+/// @param fname When NULL use the current directory name
+/// @param ext The extension to add to the filename, 4 chars max if prefixed 
+///            with a dot, 3 otherwise.
+/// @param prepend_dot If true, prefix the ext with a dot
+char *modname (char *fname, char *ext, bool prepend_dot) 
 {
-  char_u      *retval;
-  char_u      *s;
-  char_u      *e;
-  char_u      *ptr;
-  int fnamelen, extlen;
-
-  extlen = (int)STRLEN(ext);
+  char *retval;
+  size_t fnamelen;
+  size_t extlen = strlen(ext);
 
-  /*
-   * If there is no file name we must get the name of the current directory
-   * (we need the full path in case :cd is used).
-   */
+  // If there is no file name we must get the name of the current directory
+  // (we need the full path in case :cd is used).
   if (fname == NULL || *fname == NUL) {
     retval = xmalloc(MAXPATHL + extlen + 3);
```

Comment:
Once I did the effort to understand this, why not help future readers...
+ 3 is for the extra path separator, the possible (only Windows) "_", and the final NUL.
So maybe add a comment like this:
```c
    retval = xmalloc(MAXPATHL + extlen + 3);  // +3 for PATHSEP, "_" (Win), NUL
```